### ISSUE TITLE
Statistical problems with sweeps

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/msprime#1173]
It looks like there are some statistical problems in the sweeps, where we are not matching up to either analytical expectations or with msms/discoal.

@gbinux, would you mind summarising the status here please (and maybe paste in a few of the verification figures to help folks see what the problems are)?

**Comment by gbinux** [comment]
@jeromekelleher Still working on this but here are some updates:

1. It seems that when comparing `msp` and `discoal` sweeps we need to always explicitly specify `refsize` for `discoal` (the `-N` option). Otherwise the the selected allele frequency trajectory will not be correctly calculated. Internally, `discoal` uses 1000,000 as refsize by default.

2. I have compared the selected allele frequency trajectories generated by `msprime` and `discoal` side by side using the same parameters. If `refsize` is properly set, the trajectories will be comparable in terms of num of steps and delta t between steps.  So the we are good with trajectory generating function.

3. Similarly, I compared probability calculations for `p_coal_B`, `p_coal_b`, `p_rec_B`, and `p_rec_b`.  There seems to be certain probability inconsistency. My impression is that p_coal_B and p_caol_b calculated in msprime are almost always 8 times larger that that from `discoal`; `p_rec_B` and `p_rec_b` is two times larger. I simply divided `p_coal_B/b` by 8 and `p_rec_B/b` by 2 in `msprime.c` but the stats issue still persists. I think more time is needed to dig into this and these probability calculations are good starting places to work on.

**Comment by jeromekelleher** [comment]
Excellent, thanks @gbinux. Would you mind pasting in a few of the plots output by verification.py illustrating the statistical problems here please? I'm hoping to get a few more eyes on this.

**Comment by gbinux** [comment]
In the discoal original paper, a few parameters mentioned in Supplemental figure 1. With this set parameters, msms and discoal were found by the authors to generate consistent results and stats. The same set of parameters are used here to compare any two of msms, discoal and msprime for a simple selective sweeps simulation. 

The command arguments in msms format is 

`100 300 -t 40 -r 40 50000 -SF 0 0.99995 -Sp 0.5 -SaA 1000 -SAA 2000 -N 10000`

# discoal vs msprime

![image](https://user-images.githubusercontent.com/13037898/92949090-7c547380-f428-11ea-99f4-70d8c215d94d.png)

# msms vs discoal
![image](https://user-images.githubusercontent.com/13037898/92948719-d0128d00-f427-11ea-87b2-c43246765846.png)

# msms vs msprime

![image](https://user-images.githubusercontent.com/13037898/92949278-d2c1b200-f428-11ea-90be-6e7e52447bf5.png)

**Comment by andrewkern** [comment]
this must be the same time issue scaling that i was tracking down before. the time scale of the sweep phase is not appropriately matching that of the rest of the simulation. 

what are the internal units of time now in msprime? i can go look too

**Comment by gbinux** [comment]
> what are the internal units of time now in msprime?

Maybe we can run msp/discoal simulations with similar parameters and directly compare msp time and discoal time (or height of the trees). @andrewkern

**Comment by andrewkern** [comment]
yes this looks very close but not close enough yet.... i will take a look over the weekend and try to pull this together... such an annoying bug

**Comment by jeromekelleher** [comment]
Thanks for digging in to this @gbinux and @andrewkern - we're nearly there!

**Comment by andrewkern** [comment]
still working on this...

**Comment by jeromekelleher** [comment]
Can this be closed now?
